libsplinter/src/matrix.rs

```diff
@@ -12,54 +12,120 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Traits that abstract specific functionality of connection group.
+//!
+//! A component that acts as a connection group peforms the following operations
+//!   - add: Adds a connection to the underlying connection group
+//!   - remove: Removes a connection from the connection group
+//!   - send: Send a message over a specific connection
+//!   - recv: Receive a message from any ready connection in the connection group
+//!
+//! The following Matrix traits abstract out these different operations. For example an
+//! and networking component may only be interested in send and receiving messages. That component
+//! does not need to be given the methods to add or remove connection, only send and recv.
+//!
+//! Using these traits will also allow for replacing the backing connection group in the future.
+
 use std::error::Error;
 use std::fmt;
 use std::time::Duration;
 
 use crate::transport::Connection;
 
-/// Wrapper around payload to include connection id
+/// Wrapper around a payload to include connection ID
 #[derive(Debug, Default, PartialEq)]
 pub struct Envelope {
     id: String,
     payload: Vec<u8>,
 }
 
 impl Envelope {
+    /// Create a new Envelope that will be sent over a connection
```

Comment:
I think this should be changed to "Creates ..." to match the style guidelines (same with "Returns ..." for the payload method)